service/src/test/java/bio/terra/workspace/service/workspace/flight/DeleteAzureContextFlightTest.java

```diff
@@ -0,0 +1,356 @@
+
+    assertEquals(FlightStatus.SUCCESS, flightState.getFlightStatus());
+
+    // Delete the azure context.
+    FlightMap deleteParameters = new FlightMap();
+    deleteParameters.put(WorkspaceFlightMapKeys.WORKSPACE_ID, workspaceUuid.toString());
+
+    // Fail at the end of the flight to verify it can't be undone.
+    FlightDebugInfo debugInfo = FlightDebugInfo.newBuilder().lastStepFailure(true).build();
+
+    flightState =
+        StairwayTestUtils.blockUntilFlightCompletes(
+            jobService.getStairway(),
+            DeleteAzureContextFlight.class,
+            deleteParameters,
+            DELETION_FLIGHT_TIMEOUT,
+            debugInfo);
+    assertEquals(FlightStatus.FATAL, flightState.getFlightStatus());
+
+    // Because this flight cannot be undone, the context should still be deleted even after undoing.
+    assertTrue(testUtils.getAuthorizedAzureCloudContext(workspaceUuid, userRequest).isEmpty());
+  }
+
+  @Test
+  void deleteNonExistentContextIsOk() throws Exception {
+    AuthenticatedUserRequest userRequest = userAccessUtils.defaultUserAuthRequest();
+    assertTrue(testUtils.getAuthorizedAzureCloudContext(workspaceUuid, userRequest).isEmpty());
+
+    // Delete the azure context.
+    FlightMap deleteParameters = new FlightMap();
+    deleteParameters.put(WorkspaceFlightMapKeys.WORKSPACE_ID, workspaceUuid.toString());
+    deleteParameters.put(JobMapKeys.AUTH_USER_INFO.getKeyName(), userRequest);
+    FlightState flightState =
+        StairwayTestUtils.blockUntilFlightCompletes(
+            jobService.getStairway(),
+            DeleteAzureContextFlight.class,
+            deleteParameters,
+            DELETION_FLIGHT_TIMEOUT,
+            null);
+    assertEquals(FlightStatus.SUCCESS, flightState.getFlightStatus());
+    assertTrue(testUtils.getAuthorizedAzureCloudContext(workspaceUuid, userRequest).isEmpty());
+  }
+
+  @Test
+  void deleteResourcesInContext() throws Exception {
+    AuthenticatedUserRequest userRequest = userAccessUtils.defaultUserAuthRequest();
+    FlightMap createParameters =
+        azureTestUtils.createAzureContextInputParameters(workspaceUuid, userRequest);
+
+    // Create the azure context.
+    FlightState flightState =
+        StairwayTestUtils.blockUntilFlightCompletes(
+            jobService.getStairway(),
+            CreateAzureContextFlight.class,
+            createParameters,
+            CREATION_FLIGHT_TIMEOUT,
+            null);
+    assertEquals(FlightStatus.SUCCESS, flightState.getFlightStatus());
+
+    AzureCloudContext azureCloudContext =
+        workspaceService.getAuthorizedAzureCloudContext(workspaceUuid, userRequest).orElse(null);
+    assertNotNull(azureCloudContext);
+
+    // Create Azure relay namespace
+    UUID relayId = UUID.randomUUID();
+    ApiAzureRelayNamespaceCreationParameters creationParameters =
+        ControlledResourceFixtures.getAzureRelayNamespaceCreationParameters();
+    ControlledAzureRelayNamespaceResource relayNamespaceResource =
```

Comment:
🤔 wondering if there is a less time consuming resource we can spin up here, given our knowledge that this burns ~40. seconds for relay namespace creation.

Reply:
Good call, I started with a relay namespace since that's what we're creating in Rawls at the moment, but you're right that we shouldn't wait for 40 seconds here if we don't have to.